programs/mmm/src/instructions/admin/update_allowlists.rs

```diff
@@ -0,0 +1,32 @@
+use super::*;
+
+#[derive(AnchorSerialize, AnchorDeserialize)]
+pub struct UpdateAllowlistsArgs {
+    pub allowlists: [Allowlist; ALLOWLIST_MAX_LEN],
+}
+
+#[derive(Accounts)]
+#[instruction(args:UpdateAllowlistsArgs)]
+pub struct UpdateAllowlists<'info> {
+    #[account(mut, address = CANCEL_AUTHORITY)]
+    pub authority: Signer<'info>,
+    /// CHECK: Owner is not validated because this is a permissioned handler
+    pub owner: UncheckedAccount<'info>,
+    #[account(
+        mut,
+        seeds = [POOL_PREFIX.as_bytes(), owner.key().as_ref(), pool.uuid.as_ref()],
+        bump,
+        has_one = owner @ MMMErrorCode::InvalidOwner,
```

Comment:
I think you can skip checking the seeds and has_one, just the pool account is enough

Reply:
Switching this authority to be `cosigner` per our discussion means we need to validate the pool seeds and add a `has_one` constraint for the `cosigner`.